### PR TITLE
[MM-60307] Check if session is nil before calling UpdateLastActivityAtIfNeeded

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
https://github.com/mattermost/mattermost/pull/28195 missed one case. If the session linked to a `WebConn` gets invalidated, `*wc.GetSession()` panics. Avoid that by checking for `nil` first.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-60307

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by hanzei** [comment]
Heads up that I already cherry-picked this change into https://github.com/mattermost/mattermost/pull/28253 and https://github.com/mattermost/mattermost/pull/28254.

**Review by agnivade — APPROVED** [review]

File: server/channels/app/platform/web_conn.go [hunk 1]
```diff
@@ -253,7 +253,10 @@ func (ps *PlatformService) NewWebConn(cfg *WebConnConfig, suite SuiteIFace, runn
 		// Create a goroutine to avoid blocking the creation of the websocket connection.
 		ps.Go(func() {
 			ps.SetStatusOnline(userID, false)
-			ps.UpdateLastActivityAtIfNeeded(*wc.GetSession())
+			session := wc.GetSession()
+			if session != nil {
+				ps.UpdateLastActivityAtIfNeeded(*session)
+			}
```
Comment (streamer45) [inline server/channels/app/platform/web_conn.go:259]: Would simply using the original `cfg.Session` cause more problems here?

Comment (hanzei) [reply server/channels/app/platform/web_conn.go:259]: No, not using the session from `cfg. Session` was the whole point of https://github.com/mattermost/mattermost/pull/28195. At the point when `UpdateLastActivityAtIfNeeded` is executed, the session from `cfg. Session` might already be part of the `sync. Pool`, and another go routine could have ownership of it.

Comment (streamer45) [reply server/channels/app/platform/web_conn.go:259]: I see, we are indirectly using the deep copy through `wc.GetSession()`. 

@agnivade Do you feel there's still value in having a pool if we end up deep copying on every new connection?

Comment (agnivade) [reply server/channels/app/platform/web_conn.go:259]: We are deep copying only for websocket requests, not for plain HTTP requests. So think a pool still makes sense.

File: server/channels/app/platform/session.go [hunk 2]
```diff
@@ -21,7 +21,7 @@ func (ps *PlatformService) ReturnSessionToPool(session *model.Session) {
 		// To avoid a race between clearing the props and accessing it, clear the props maps before returning it to the pool.
 		clear(session.Props)
 		// Also clear the team members slice to avoid a similar race condition.
-		clear(session.TeamMembers)
+		session.TeamMembers = session.TeamMembers[:0]
```
Comment (agnivade) [inline server/channels/app/platform/session.go:24]: @hanzei - I thought we decided on removing the TeamMembers clearing entirely?

**Review by agnivade — APPROVED** [review]

File: server/channels/app/platform/session.go (hunk 2 above)
Comment (hanzei) [reply server/channels/app/platform/session.go]: Applied the change

**Pull request merged**